ang/sqlTaskConfigurator.js

```diff
@@ -576,6 +578,230 @@
+                  ([ value, name ]) => ({ value, name })
+                )
+              );
+            });
+          }).catch(error => {
+            console.error(error);
+            return [];
+          });
+        }
+
+        async function initialSetup() {
+          let entity = $scope.model.entity;
+
+          $scope.apiv4Entities = await fetchEntities();
+          $scope.apiv4ErrorHandlerOptions = await fetchErrorHandlerOptions();
+          $scope.$apply();
+
+          if (!isValidEntity(entity)) {
+            entity = undefined;
+          }
+
+          await selectEntity(entity);
+          await updateAction(entity);
+        }
+
+        function isValidEntity(entity) {
+          return $scope.apiv4Entities.find(({ name }) => name === entity) !== undefined;
+        }
+
+        function parseApiExplorerUrl(urlString) {
+          const result = {
+            entity: undefined,
+            action: undefined,
+            parameters: {},
+          }
+
+          try {
+            const url = new URL(urlString);
+
+            let urlHash = url.hash;
+            let matches = urlHash.match(/^#\/explorer(?<urlHash>\/.*)?$/);
+
+            if (matches === null) throw new Error(`Invalid URL hash: '${urlHash}'`);
+
+            urlHash = matches.groups.urlHash;
+
+            if (urlHash === undefined) return result;
+
+            matches = urlHash.match(/^\/(?<entity>[a-zA-Z0-9_]+)?(?<urlHash>\/.*)?$/);
+            result.entity = matches.groups.entity;
+            urlHash = matches.groups.urlHash
+
+            if (urlHash === undefined) return result;
+
+            matches = urlHash.match(/^\/(?<action>[a-zA-Z0-9_]+)?(?<urlHash>\?.*)?$/);
+            result.action = matches.groups.action;
+            urlHash = matches.groups.urlHash
+
+            if (urlHash === undefined) return result;
+
+            result.parameters = parseApiCallParams(urlHash);
+          } catch (error) {
+            console.error("Invalid APIv4 Explorer URL:", urlString);
+            console.error(error);
+          }
+
+          return result;
+        }
+
+        function parseApiCallParams(searchParamsStr) {
+          const parameters = {};
+          const searchParams = new URLSearchParams(searchParamsStr);
+
+          for (const [key, value] of searchParams.entries()) {
+            switch (key) {
+              case "limit": {
+                parameters.limit = parseInt(value, 10);
+                break;
+              }
+
+              case "groupBy":
+              case "having":
+              case "join":
+              case "select":
+              case "where": {
+                parameters[key] = JSON.parse(value);
+                break;
+              }
+
+              case "chain":
+              case "orderBy":
+              case "values": {
+                parameters[key] = Object.fromEntries(JSON.parse(value));
+                break;
+              }
+
+              default: {
```

Comment:
Do you think we can add a generic parameter handler here? Some API4 actions can define their own (non-standard) fields, e.g. in `mailutils` we have `MailutilsMessage.send`, which accepts `messageId`.

Perhaps it would be enough to run it through `JSON.parse()`; that'd handle integers and (quoted) strings just as well. We wouldn't be able to know whether it's a `where` or `values` kind of thing for arrays, but still better than nothing.

(Maybe we can ~steal~ use some code from `ang/api4Explorer/Explorer.js` (in core) related to this; API Explorer would need to do the same thing in reverse when generating the JS sample code.)

Reply:
As far as I know `chain`, `orderBy` and `values` are the only parameters that need this kind of array-to-object conversion anyway, so `JSON.parse` should cover everything else.

Reply:
Looks good and works now.

I found a new/semi-related thing while testing this:
In newer versions of Civi, the API Explorer will add a `_lang=js` parameter to the URL if you switch to e.g. the auto-generated JS code at the bottom. This seems to break parameter parsing in general, i.e. it just produces `{}`.

I think the idea is that any parameter starting with underscore can be ignored (i.e. is API Explorer UI-related only; there might be more than just `_lang`).

Example URL: `http://civicrm.example.com/civicrm/api4#/explorer/MailutilsMessage/send?messageId=123&checkPermissions=0&_lang=js`